src/app/cli/interpreter.cpp

```diff
@@ -1443,9 +1444,19 @@ Interpreter::Value Interpreter::ProcessBr(const Expression &aExpr)
             }
         }
 
+        if (mContext.mCommandKeys.size() == 2 && mContext.mCommandKeys[0] == "--netif")
+        {
+            netIf = mContext.mCommandKeys[1];
+        }
+
         // Open IPv4 mDNS socket
         mdnsSocket = mdns_socket_open_ipv4();
         VerifyOrExit(mdnsSocket >= 0, value = ERROR_IO_ERROR("failed to open mDNS IPv4 socket"));
+
+        if (netIf != "" && setsockopt(mdnsSocket, SOL_SOCKET, SO_BINDTODEVICE, &netIf[0], sizeof(netIf)) < 0) {
+          ExitNow(value = ERROR_IO_ERROR("failed to bind network interface: {}", netIf));
```

Comment:
Please add the error code to the returned value so that the user knows what's wrong here